File: app/src/main/java/com/aquamorph/frcmanager/fragments/setup/Slide.java
package com.aquamorph.frcmanager.fragments.setup;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * Generic slide fragment that can be passed a layout.
 *
 * @author Christian Colglazier
 * @version 3/29/2016
 */
public class Slide extends Fragment {

	private static final String ARG_LAYOUT_RES_ID = "layoutResId";
	private int layoutResId;

	public static Slide newInstance(int layoutResId) {
		Slide sampleSlide = new Slide();
		Bundle args = new Bundle();
		args.putInt(ARG_LAYOUT_RES_ID, layoutResId);
		sampleSlide.setArguments(args);
		return sampleSlide;
	}

	@Override
	public void onCreate(@Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		if (getArguments() != null && getArguments().containsKey(ARG_LAYOUT_RES_ID)) {
			layoutResId = getArguments().getInt(ARG_LAYOUT_RES_ID);
		}
	}

	@Nullable
	@Override
	public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
	                         @Nullable Bundle savedInstanceState) {
		return inflater.inflate(layoutResId, container, false);
	}
}

File: app/src/main/kotlin/com/aquamorph/frcmanager/fragments/setup/Slide.kt
package com.aquamorph.frcmanager.fragments.setup

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment

/**
 * Generic slide fragment that can be passed a layout.
 *
 * @author Christian Colglazier
 * @version 3/31/2018
 */
class Slide : Fragment() {
    private var layoutResId: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null && requireArguments().containsKey(ARG_LAYOUT_RES_ID)) {
            layoutResId = requireArguments().getInt(ARG_LAYOUT_RES_ID)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(layoutResId, container, false)
    }

    companion object {

        private const val ARG_LAYOUT_RES_ID = "layoutResId"

        /**
         * newInstance() creates a new slide.
         *
         * @return slide
         */
        fun newInstance(layoutResId: Int): Slide {
            val slide = Slide()
            val args = Bundle()
            args.putInt(ARG_LAYOUT_RES_ID, layoutResId)
            slide.arguments = args
            return slide
        }
    }
}
